scripts/train.py

```diff
@@ -1,34 +1,60 @@
 import argparse
+import logging
 import os
 import joblib
 import pandas as pd
-from house_package_saachi.data_ingestion import fetch_housing_data, load_housing_data
+from house_package_saachi.data_ingestion import load_housing_data
 from house_package_saachi.data_preprocessing import preprocess_housing_data
 from house_package_saachi.model_training import train_linear_regression, evaluate_model
 
+def setup_logger(log_level, log_path=None, no_console_log=False):
+    logger = logging.getLogger(__name__)  # Ensure the logger uses the module's name
+    logger.setLevel(log_level)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+    if log_path:
+        file_handler = logging.FileHandler(log_path)
+        file_handler.setFormatter(formatter)
+        logger.addHandler(file_handler)
+
+    if not no_console_log:
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(formatter)
+        logger.addHandler(console_handler)
+
 def main(input_path, output_path):
+    logger = logging.getLogger(__name__)  # Get the logger for this module
+
+    logger.info("Loading housing data.")
     housing = load_housing_data(input_path)
     
-    # Separate the target from the features
+    logger.info("Separating target from features.")
     housing_labels = housing["median_house_value"].copy()
     housing = housing.drop("median_house_value", axis=1)
     
+    logger.info("Preprocessing housing data.")
     housing_prepared = preprocess_housing_data(housing)
     
-    # Train the model
+    logger.info("Training the linear regression model.")
     lin_reg = train_linear_regression(housing_prepared, housing_labels)
     
-    # Evaluate the model
+    logger.info("Evaluating the model.")
     rmse = evaluate_model(lin_reg, housing_prepared, housing_labels)
-    print(f"Training RMSE: {rmse}")
+    logger.info(f"Training RMSE: {rmse}")
     
-    # Save the model
+    logger.info("Saving the trained model.")
     os.makedirs(output_path, exist_ok=True)
```

Comment:
@saachi-tiger i forgot to mention it previously..
can you add some of the logging statements from inside the functions i.e in functions written inside `src/house_package_saachi/`

Reply:
Added few logging statements in functions